### PR TITLE
migrate libbeat to static

### PR DESCRIPTION
**Pull request by sharbuz**
## What is the problem this PR solves?

Jenkins->Buildkite pipelines migration 

Migrate libbeat from dynamic to static pipeline

Example of the pipelines:
The main pipeline: https://buildkite.com/elastic/beats/builds/5588
The libbeat pipeline: https://buildkite.com/elastic/beats-libbeat/builds/3772

## Related issues
https://github.com/elastic/ingest-dev/issues/1693

### CODE REVIEW

File: .buildkite/libbeat/pipeline.libbeat.yml [hunk 1]
```diff
@@ -4,50 +4,81 @@ name: "beats-libbeat"
...
+          provider: "gcp"
+          image: "${IMAGE_UBUNTU_X86_64}"
+          machineType: "${GCP_HI_PERF_MACHINE_TYPE}"
+        artifact_paths:
+          - "$BEATS_PROJECT_NAME/build/*.xml"
+          - "$BEATS_PROJECT_NAME/build/*.json"
+
+      - label: ":testengine: Stress Tests"
+        key: "mandatory-stress-test"
+        command: "cd $BEATS_PROJECT_NAME && make STRESS_TEST_OPTIONS='-timeout=20m -race -v -parallel 1' GOTEST_OUTPUT_OPTIONS='| go-junit-report > libbeat-stress-test.xml' stress-tests"
```
Comment (dliappis) [inline .buildkite/libbeat/pipeline.libbeat.yml:64]:
```suggestion
        command: |
          set -euo pipefail
          cd libbeat
          make STRESS_TEST_OPTIONS='-timeout=20m -race -v -parallel 1' GOTEST_OUTPUT_OPTIONS=' | go-junit-report > libbeat-stress-test.xml' stress-tests
```

**Review by dliappis — COMMENTED** [review]
Left a few comments also based on offline discussions

**Review by dliappis — APPROVED** [review]
LGTM

**Review by jlind23 — APPROVED** [review]

**Pull request merged**